common/scala/src/main/scala/whisk/core/entity/ExecManifest.scala

```diff
@@ -91,12 +105,93 @@ protected[core] object ExecManifest {
      */
     protected[entity] case class RuntimeManifest(
         kind: String,
+        image: ImageName,
         deprecated: Option[Boolean] = None,
         default: Option[Boolean] = None,
         attached: Option[Attached] = None,
         requireMain: Option[Boolean] = None,
-        sentinelledLogs: Option[Boolean] = None,
-        image: Option[String] = None) {
+        sentinelledLogs: Option[Boolean] = None) {
+
+        protected[entity] def toJsonSummary = {
+            JsObject(
+                "kind" -> kind.toJson,
+                "deprecated" -> deprecated.getOrElse(false).toJson,
+                "default" -> default.getOrElse(false).toJson,
+                "attached" -> attached.isDefined.toJson,
+                "requireMain" -> requireMain.getOrElse(false).toJson,
+                "image" -> image.publicImageName.toJson)
+        }
+    }
+
+    /**
+     * An image name for an action refers to the container image canonically as
+     * "prefix/name[:tag]" e.g., "openwhisk/python3action:latest".
+     */
+    protected[entity] case class ImageName(
+        name: String,
+        prefix: Option[String] = None,
+        tag: Option[String] = None) {
+
+        /**
+         * The name of the public image for an action kind.
+         */
+        def publicImageName: String = {
+            val p = prefix.filter(_.nonEmpty).map(_ + "/").getOrElse("")
+            val t = tag.filter(_.nonEmpty).map(":" + _).getOrElse("")
+            p + name + t
+        }
+
+        /**
+         * The internal name of the image for an action kind. It overrides
+         * the prefix with an internal name. Optionally overrides tag.
+         */
+        def localImageName(registry: String, prefix: String, tagOverride: Option[String] = None): String = {
+            val r = Option(registry).filter(_.nonEmpty).map { reg =>
+                if (reg.endsWith("/")) reg else reg + "/"
+            }.getOrElse("")
+            val p = Option(prefix).filter(_.nonEmpty).map(_ + "/").getOrElse("")
+            r + p + name + ":" + tagOverride.orElse(tag).getOrElse(ImageName.defaultImageTag)
+        }
+
+        /**
+         * Overrides equals to allow match on undefined tag or when tag is latest
+         * in this or that.
+         */
+        override def equals(that: Any) = that match {
+            case ImageName(n, p, t) =>
+                name == n && p == prefix && (t == tag || {
+                    val thisTag = tag.getOrElse(ImageName.defaultImageTag)
+                    val thatTag = t.getOrElse(ImageName.defaultImageTag)
+                    thisTag == thatTag
+                })
+
+            case _ => false
+        }
+    }
+
+    protected[entity] object ImageName {
+        protected val defaultImageTag = "latest"
+        private val componentRegex = """([a-z0-9._-]+)""".r
+        private val tagRegex = """([\w.-]{0,128})""".r
```

Comment:
Do we really need to have those narrow rules in here? After all, we're copying docker's regexes are we? What do we gain by this?